app/controllers/posts_controller.rb

```diff
@@ -1,11 +1,38 @@
 class PostsController < ApplicationController
+  before_action :set_post, only: %i[show edit update destroy]
+
   def index
     @user = User.find(params[:user_id])
-    @post = @user.posts
+    @posts = @user.posts
   end
 
   def show
     @user = User.find(params[:user_id])
-    @post = @user.posts.find(params[:post_id])
+  end
+
+  def new
+    @post = Post.new(author_id: params[:user_id], comments_counter: 0, likes_counter: 0)
+  end
+
+  def create
+    @post = Post.new(post_params)
+
+    respond_to do |format|
+      if @post.save
+        format.html { redirect_to user_posts_path, notice: 'Post was successfully created.' }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+      end
+    end
+  end
```

Comment:
- [ ] Following the project requirements, a post should be created on behalf of the currently logged-in user (current_user) of the app. Please use the current_user method that you created in the application_controller to create a post rather than using params[:user_id].